tests/ashley/test_forum_search.py

```diff
@@ -730,3 +731,80 @@ def test_forum_search_paginated_links(self):
             )
 
         self.assertNotContains(response, "?q=a5g3&amp;page=7")
+
+    def test_forum_search_show_direct_post_with_right_pagination(self):
+        """
+        Check that the link of the results targets directly the post in
+        the topic. Using this link, the user get directed at the right
+        page of the topic the post is related to.
+        """
+
+        user = UserFactory()
+        post = PostFactory(text="azatofind")
+        PostFactory.create_batch(49, topic=post.topic)
+        # Create a specific post between two batchs
+        post_fourth_page = PostFactory(text="azatofind", topic=post.topic)
+        PostFactory.create_batch(50, topic=post.topic)
+        post_last_page = PostFactory(text="azatofind", topic=post.topic)
+        assign_perm("can_read_forum", user, post.topic.forum)
+
+        # Index posts in Elasticsearch
+        call_command("rebuild_index", interactive=False)
+
+        self.client.force_login(user)
+        response = self.client.get("/forum/search/?q=azatofind")
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(
+            response, "Your search has returned <b>3</b> results", html=True
+        )
+        url_topic = (
+            f"/forum/forum/{post.topic.forum.slug}-"
+            f"{post.topic.forum.id}/topic/{post.topic.slug}-{post.topic.id}/"
+        )
+
+        direct_link_initial_post = f"{url_topic}?post={ post.pk }#{ post.pk }"
+        direct_link_post_fourth_page = (
+            f"{url_topic}?post={ post_fourth_page.pk }#{ post_fourth_page.pk }"
+        )
+        direct_link_post_last_page = (
+            f"{url_topic}?post={ post_last_page.pk }#{ post_last_page.pk }"
+        )
+
+        # links of three posts are present
+        self.assertContains(response, direct_link_initial_post)
+        self.assertContains(response, direct_link_post_fourth_page)
+        self.assertContains(response, direct_link_post_last_page)
+
+        # confirm the last page of the topic is page 7
+        response = self.client.get(url_topic)
+        self.assertContains(response, '<a href="?page=7" class="page-link">7</a>')
+        self.assertNotContains(response, '<a href="?page=8" class="page-link">8</a>')
+
+        response = self.client.get(direct_link_initial_post)
+        # first post is at page 1
+        self.assertContains(
```

Comment:
Not sure I understood correctly but calling the view with a "post" parameter in the querystring will redirect to the correct page at which the post is? If yes can you make it more explicit in the docstring by using the term "redirect"?

Using the Django redirect test would also make it clearer: https://docs.djangoproject.com/en/3.0/topics/testing/tools/#django.test.SimpleTestCase.assertRedirects

Reply:
no it doesn't get redirected, it just loads the page directly at the right pagination. I've rephrased it to be more accurate.

Reply:
oh :astonished:  :+1: